### DISCUSSION TITLE
Probleme mit dem WLAN-Repeater-Update

### DISCUSSION THREAD

**Discussion by asmcc** [Freetz-NG/freetz-ng#850] [Allgemein] [answered]
Ich habe heute meinen WLAN-Repeater 1750E auf die neue FREETZ-NG mit der 31-Firmare "hochgezogen". Die Repeater waren schon immer etwas "zickiger", was das Update angeht, diesmal hat es sich aber wieder bestätigt. Dabei ist mir folgendes Verhalten aufgefallen:
Der Repeater lies sich mit einem etwas älteren FREETZ (Anfang 2023) nicht über AVM-WebIF flashen. Es kamen die E-Mails: Update gestartet / Update abgebrochen. Auch über FREETZ-Update mit Standard-Einstellungen lief das Update anscheinend in einenWatchdog und die Box rebootete am Ende ohne Update gemacht zu haben.
![20230917_Update_failed_1750E](https://github.com/Freetz-NG/freetz-ng/assets/40981206/934789b1-34a4-498f-b741-b84f148bf224)

Abhilfe habe ich mir nur über die Option "alle Dienste anhalten (bei Speichermangel)" verschaffen. Danach konnte ich über FREETZ-WebIF-Update erfolgreich flashen.
![grafik](https://github.com/Freetz-NG/freetz-ng/assets/40981206/74f9cbf5-7ce6-4bc6-ac1f-a5eae242bc39)
![grafik](https://github.com/Freetz-NG/freetz-ng/assets/40981206/45a3b842-d3a2-4ff7-a449-b3b299297df9)

Nun habe ich ein weiteres Image erstellt mit etwas mehr Funktionen und der Zirkus mit dem Updaten ging wieder von vorne los. Zu einem habe ich bei FREETZ-WebIF-Update keine Möglichkeit mehr gefunden, dieses "alle Dienste anhalten (bei Speichermangel)" zu aktivieren. Und ohne diese Option läuft das Update über FREETZ WebIF schief und endet in einem Watchdog-Reboot ohne Update.
Erstaunlicherweise klappte danach aber das Update über AVM-WebIF. Somit bin ich am Ende natürlich glücklich. Dennoch würden mich schon ein Paar Details interessieren:
1. Wurde in den letzten Monaten am Update-Verhalten was angepasst? Ich verfolge leider FREETZ-NG nur sporadisch.
2. Kann man die alte Option "alle Dienste anhalten" in FREETZ-WebIF-Update bitte wieder aufnehmen oder gibt es irgendein Trick, wie man es aktivieren kann?

Danke!

**Comment by fda77** [comment]
* 3eff77c35 webif flash: service stopping only for old devices 

Das wurde deaktiviert da es in einem Forum grundsätzlich empfohlen wurde dies zu setzen. Es sollte aber nur genutzt werden wenn wirklich nötig.  Ich würde dir auch empfehlen lieber ein paar Freetz Dienste die viel ram brauchen vorher zu beenden.
Sonst kann es sein wenn das Flashen zu lange dauert der watchdoch mittendrin rebootet und es "nicht geht"

* 79ecb1d6d webif flash: service stopping only if less than 100mb ram

**Reply by asmcc** [reply]
Ich glaube nicht, dass es damit zu tun hat, cuma, dass er auf irgendein Dienst wartet, um es zu beenden. Das Protokoll ist da ja auch leer (s. oben im Bild). Nach einem RAM-Überlauf sieht es ebenfalls für mich nicht aus. Die RAM-Überläufe kenne ich von den anderen Boxen. Die passieren deutlich schneller. Ich pinge ja die Box parallel an, um zu sehen, wann sie weg ist. Und das sieht definitiv nach einem sehr langem Watchdog aus. Ich schätze 60 oder 90 Sekunden, gefühlt. Da ist irgendein weiterer Wurm drin, speziell bei den Repeatern. Das ist meine Vermutung.
Natürlich beende ich pro aktiv alle Dienste, die es nur geht zu beenden, bevor ich flashe. Nach meinen >15 Jahren hier weiß ich das alles schon. Das ist auch der Grund, warum ich bis jetzt FREETZ-WebIF Update dem AVM-Update bevorzuge. Ich wollte hier lediglich signalisieren, dass Repeater eben Speziallfälle sind und öfters "aus der Reihe" tanzen. Aber danke, dass du mir gezeigt hast, wo es deaktivert wurde. Vielleicht reaktiviere ich es in meinem eigenen Fork und teste, ob es was bringt und das Flashen über FREETZ WebIF somit dann wieder funktioniert.
Komisch ist für mich eben, dass es mit dem Stand von Januar genau andersrum nicht ging: Ich konnte über AVM WebIF nicht flashen. Und jetzt kann ich plötzlich über AVM WebIF flashen und unter FREETZ WebIF nicht, weil da dieser "Schalter" fehlt.
Und dieses Verhalten ist genau so reproduzierbar, wie ich es beschreibe. Ich habe hier zwei von solchen Repeatern, die sich genau gleich verhalten haben.
Aber gut, du hast vermutlich keine 1750E zur Hand, um es zu reproduzieren und so viele User gibt es hier auch nicht, die so einen zickigen Repeater überhaupt modifizieren wollen. Von daher bleibt es vermutlich mein Problem.

**Reply by asmcc** [reply]
Achso, ich habe erst jetzt verstanden, dass du da jetzt doch was vor 38 Minuten geändert hast...
Ich teste es dann die Tage mit deinem neuen commit. Aber ja, dieser pre-supervisor kann da das Problem sein. Den gibt es bei 1750E tatsächlch (also wird dann die -z Prüfung fehlschlagen), obwohl 1750E weniger als 100MB RAM hat. Von daher sollte ich nun diesen Menüpunkt mit deinem neuen commit zu sehen bekommen. Zumindest bei dem 1750E-Repeater.

**Comment by fda77** [comment, marked as answer]
Genau, der 1. hat es für deine Box weggemacht und der 2. commit sollte es wieder sichbar gemacht haben
Rechne doch mal selbst mit deinen Daten:
```
64mb ram
 - die firmware die du (entpackt) ins ram zum flashen hochlädst
 - das was avm belegt
 - die zusätzlichen freetz services
 = OOM :)
```

**Reply by asmcc** [reply]
Also, mit dem neuen commit bekommt man es zu sehen. Allerdings hatte ich eine weitere Beobachtung, die deine Theorie voll bestätigt, cuma. Ich konnte diesmal per AVM Web IF nicht flashen. Bzw. er hat angeblich geflasht, es kam auch die E-Mail "Ihr System wurde erfolgreich upgedatet", auf der Box war aber am Ende das alte Image immer noch. Das hatte ich 3x wiederholt und dann damit aufgegeben und musste etwas tricksen. Mit dem "mount -o bind" die besagte Datei "überschrieben", sodass ich dann aus FREETZ WebIF flashen konnte und mit diesem "Umweg" dann geflasht. Warum es jetzt mit dem AVM Update nicht geklappt hat und vor allem warum er dann trotzdem "erfolgreich" gemeldet hatte, ist für mich ein Rätsel. Ich vermute da aber mittlerweile genau das, worauf du zielst: Die Größe des Images liegt genau an der Grenze des freien Platzes im RAM. Wenn man alle Dienste runterfährt, reicht es dann vermutlich gerade so. Wenn ein Paar übrig bleiben, dann wird es grenzwertig und dann passt es mal und mal nicht.

**Reply by fda77** [reply]
Das Gerät ist vom Ram her am Ende, ich hatte vor längerem auch mal so einen. Der fast Baugleiche DVBC hat noch weniger Ram frei wegen dem ganzen DVB kram
Aber im Zweifel kann man immer über den Bootloader (zb push_firmware) flashen, da ist das egal.
Statt dem mount-bind etc gibt es das Wrapper "vix" bzw jetzt neu vimx und nanox

**Reply by asmcc** [reply]
ja, scheint hier der Fall zu sein, reicht aber als WLAN-Repeater gerade aus, ist relativ "pflegeleicht" sonst (abgesehen vom Updaten), kann in die Steckdose eingesteckt werden, unterstützt MESH. Ich hatte die irgendwann mal vor ein Paar Monaten günstig ersteigert. Jemand hatte seine 4 Stück verkauft, weil er damit nicht klar kam, oder wie auch immer. Ich habe sie nun in alle Ecken des Hauses verteilt (Garage usw.) und somit die WLAN-Abdeckung im Haus deutlich verbessert.
Danke für die Tipps mit dem vix, kannte ich nicht bis jetzt. Teste irgendwann mal. Wobei ich kein Fan von vi bin und nano wird da nicht rein passen.

**Reply by fda77** [reply]
Falls man für das Gerät Module bauen kann, könntest du autofs nehmen um damit einen nfs server zu mounten. Und external kann von dort Packages einbinden. Da hatte ich mit älteren Modellen mal so laufen
